rules/windows/credential_access_credential_dumping_msbuild.toml

```diff
@@ -12,34 +12,33 @@ credential management. This technique is sometimes used for credential dumping.
 false_positives = ["The Build Engine is commonly used by Windows developers but use by non-engineers is unusual."]
 from = "now-9m"
 index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
-language = "kuery"
+language = "eql"
 license = "Elastic License v2"
 name = "Microsoft Build Engine Loading Windows Credential Libraries"
 risk_score = 73
 rule_id = "9d110cb3-5f4b-4c9a-b9f5-53f0a1707ae5"
 severity = "high"
 tags = ["Elastic", "Host", "Windows", "Threat Detection", "Credential Access"]
 timestamp_override = "event.ingested"
-type = "query"
+type = "eql"
 
 query = '''
-event.category:process and event.type:change and
-  (process.pe.original_file_name:(vaultcli.dll or SAMLib.DLL) or
-    dll.name:(vaultcli.dll or SAMLib.DLL)) and
-  process.name: MSBuild.exe
+sequence by process.entity_id
+ [process where event.type == "start" and (process.name : "MSBuild.exe" or process.pe.original_file_name == "MSBuild.exe")]
+ [process where event.type == "change" and dll.name : ("vaultcli.dll", "SAMLib.DLL")]
```

Comment:
can you add the `process.name` here too?
Elasticsearch only knows to copy that constraint if it's part of a join `by` key, so that's an option too

Reply:
what does `change` mean for an event.type on windows? that doesn't sound right.

maybe we should do one thing at a time and go flat instead of a sequence

Reply:
`change` will mean an event.action of `modification`, `overwrite` or `rename`. i thought that was for file and registry events only though.

Reply:
now it will look for msbuild.exe followed by imageload of vaultcli or samlib which matches the description of the rule.